### PR TITLE
[Sample] Improve tfx oss sample

### PR DESCRIPTION
**Pull request by numerology**
Did two things here:
- Fix the flakiness discovered in #2354 
- Improve how the default value of pipeline-root look like by not passing it in its string form.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/2360)

### CODE REVIEW

File: test/sample-test/run_sample_test.py [hunk 1]
```diff
@@ -111,7 +112,9 @@ def run(self):
 
     # TODO(numerology): Special treatment for TFX::OSS sample
     if self._testname == 'parameterized_tfx_oss':
-      self._test_args['pipeline-root'] = os.path.join(self._test_args['output'], 'tfx_taxi_simple')
+      self._test_args['pipeline-root'] = os.path.join(
+          self._test_args['output'],
+          'tfx_taxi_simple_' + str(int(time.time())))
```
Comment (Ark-kun) [inline test/sample-test/run_sample_test.py]: You might also use the `kfp.dsl.RUN_ID_PLACEHOLDER`.

Comment (numerology) [reply test/sample-test/run_sample_test.py]: Good idea. I think that's better

Comment (numerology) [reply test/sample-test/run_sample_test.py]: @Ark-kun  Done.

**Pull request merged**